### ISSUE TITLE
Is finding 68 points with DLib  or deep neural networks?

### ISSUE THREAD

**Issue by mahna3411** [TadasBaltrusaitis/OpenFace#867]
hi , Thanks for the openface

Is finding points (68 points) in the OpenFace project done by DLib model or deep neural networks?

I implemented this in Unity C#
But I can only test the similarity of the face.

I'm looking for a way to implement face recognition with deep neural networks. 

Please help me thank you

**Comment by TadasBaltrusaitis** [comment]
It is using the CE-CLM algorithm ([link to paper](https://arxiv.org/abs/1611.08657)) for facial landmark detection, it is not using dlib. It can use a dlib face detector as a first step in this pipeline.

I'm not entirely sure I understand your question?

**Comment by mahna3411** [comment]
After running it on C #, I got a 128-dimensional matrix.
How do I extract facial marks from it?

**Comment by TadasBaltrusaitis** [comment]
Can you be a bit more specific, after running what on C#?

**Comment by mahna3411** [comment]
yes, 
My problem is this:
I can compare the similarity in the end with the C # in openface
but 

In fact, the implementation I did on C# is similar to the JavaScript code below:
https://github.com/opencv/opencv/blob/master/samples/dnn/js_face_recognition.html


How can I extract facial landmark from a 128-dimensional matrix?

![photo_2020-05-04_17-39-40](https://user-images.githubusercontent.com/45923789/80969102-4fff6980-8e2e-11ea-876c-c505a2885bbf.jpg)

**Comment by TadasBaltrusaitis** [comment]
I'm not sure the 128-dimensional vector/matrix is coming from OpenFace. What function returns it?

This seems like a face recognition vector that can be used to measure face similarities. There is no such vector as part of OpenFace output.

**Comment by mahna3411** [comment]
Mat TargetMat = net02.forward();

Does this mean that:
dnn model (nn4.small2) is not able to find 68 points on the face and this is done with dlib(shape_predictor_68_face_landmarks.bytes) in openFace project (face recognition with deep neural networks)?

**Comment by TadasBaltrusaitis** [comment]
Are you sure you are using OpenFace from this repository? Confusingly, there is an OpenFace project for face recognition (https://cmusatyalab.github.io/openface/). while this project is for facial expression, facial landmark, head pose and eye gaze tracking.

**Comment by mahna3411** [comment]
Yes, I have used this project, but in another programming language, my question is:
facial landmark, head pose and eye gaze tracking
Are these extracted from this nn4.small2 model?

**Comment by TadasBaltrusaitis** [comment]
I'm not sure what the network you are referring to is, I don't think it is part of OpenFace, so am not able to answer your question. Are you sure you are referring to this OpenFace, as I don't know what nn4.small2 is, can you provide the exact code?

**Comment by mahna3411** [comment]
this is openFace Model 
![Capture00](https://user-images.githubusercontent.com/45923789/81435240-c7116680-917c-11ea-8893-6898488d8fa9.JPG)

**Comment by TadasBaltrusaitis** [comment]
That's the other OpenFace project that I mentioned, that does face recognition.

This OpenFace project is intended for landmark, head pose, eye gaze and facial expression recognition.

**Comment by mahna3411** [comment]
Send the link to this project that you mentioned, thank you.

**Comment by TadasBaltrusaitis** [comment]
The link to the other project is here - https://cmusatyalab.github.io/openface/
